packages/components/src/styles/components/code-block/index.scss

```diff
@@ -262,3 +294,60 @@ $hds-code-block-code-padding: 16px;
     font-style: italic;
   }
 }
+
+// Footer
+.hds-code-block__overlay-footer {
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 10px $hds-code-block-code-padding 10px $hds-code-block-code-padding;
+  pointer-events: none; // fix issue with scrolling when hovering over the footer
+
+  // re-enable pointer events for the button (or any content inside the footer)
+  > * {
+    pointer-events: auto;
+  }
+}
+
+.hds-code-block--has-overlay-footer {
+  .hds-code-block__code::after {
+    // gradient element
+    position: sticky; // prevent gradient from scrolling together with content
+    bottom: -$hds-code-block-code-padding;
+    left: 0;
+    display: block;
+    height: $hds-code-block-code-footer-height;
+    margin: 0 0 -#{$hds-code-block-code-padding} -1000px;
```

Comment:
Might be nice to add a comment here to remember why we set these values specifically.

Reply:
Probably also link to the slack thread(s), just in case (there's a lot of info on what we tried, what worked/didn't work, etc)